hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconPipelineManager.java

```diff
@@ -104,12 +104,13 @@ void initializePipelines(List<Pipeline> pipelinesFromScm)
       List<Pipeline> pipelinesInHouse = getPipelines();
       LOG.info("Recon has {} pipelines in house.", pipelinesInHouse.size());
       for (Pipeline pipeline : pipelinesFromScm) {
-        if (!containsPipeline(pipeline.getId())) {
-          // New pipeline got from SCM. Recon does not know anything about it,
-          // so let's add it.
-          LOG.info("Adding new pipeline {} from SCM.", pipeline.getId());
-          addPipeline(pipeline);
+        // New pipeline got from SCM. Recon does not know anything about it,
```

Comment:
This above comments seems not fit until we find out if Recon contains that pipeline or not.
```suggestion
        // New pipeline got from SCM. Validate If it doesn't exists at Recon, try adding it.
```